tcping.go

```diff
@@ -87,7 +88,7 @@ func signalHandler(tcpStats *stats) {
 
 	go func() {
 		<-sigChan
-		tcpStats.endTime = time.Now()
+		tcpStats.endTime = tcpStats.startTime.Add(tcpStats.totalUptime).Add(tcpStats.totalDowntime)
```

Comment:
This probably should be done like this to 100% match other results.
```suggestion
tcpStats.startTime.Add((totalUnsuccessfulProbes + totalUnsuccessfulProbes) * time.Second)
```